src/compile/scale/type.ts

```diff
@@ -77,6 +78,8 @@ function defaultType(channel: Channel, fieldDef: TypedFieldDef<string>, mark: Ma
         log.warn(log.message.discreteChannelCannotEncode(channel, 'temporal'));
         // TODO: consider using quantize (equivalent to binning) once we have it
         return 'ordinal';
+      } else if (fieldDef.timeUnit && isUTCTimeUnit(fieldDef.timeUnit)) {
+        return 'utc';
```

Comment:
Why changing scale type? Wont this leads to double shift that we discussed earlier?

Reply:
Playing around with [this spec](https://vega.github.io/editor/#/url/vega/N4KABGBEAkDODGALApgWwIaQFxUQFzwAdYsB6UgN2QHN0A6agSz0QFcAjOxge1IRQyUa6SgFY6AK1jcAdpAA04KABNkCAE6NCeHnJyQA4t27UANsgDksMLDzd4AazCFN8ZGG5V1YHamR0FJUh2dEdqdW5WGWVsKAB3RGZkQIhIQnRlZUYZaljRRVS4xmUWWIAmAAYKgtxkRmp8cqqayFsAT3NYyDdTUxSVdDxMHABtJQhQCCmoGXQ-LulWdTcAfQr+6cgKdFNWNVix6engSGVB5P0AZSj5MAqARjAAKXQZMEr7sveAZiwqv-WtzSrguYHuogqAF8akcwCcznhQZBrjJbhUvi83h8vv9cYCoC5GG5YvcqtDxkd4ecuii0RjXu8Hjj7gCAQoCSCSQB2KEwymnalXG53elYpki1lVdnAomg+4ANlE5NhEwFiJpwvRzwZ2Luvzx0sJxJwlShFIgAF0KcqppMjpBZvN9Aj0GsNlNWpFlkjFt63XzUnh1K9YAAzbjqVAHc22yB4NqEJHhyOsUyYIHIAAeLi6dgAIucABQI1ioEYAHTVyErFoAlNL0LAugjkjbYXaVVBQ4xkKYYs7qQH7fHE7nGH4osx3bC4+PkAAvWRI1h4eDT+2TvBN0aQNrIdDqaWoWSlIEt6WIL1NoGobIr-YWoebRvRzubPcH48yFgty9LWC3jI96wCs55PvaH6RieiC-legHAaB5wrMg0SQDG0xWp2kIxphUzYZaLQYOoDjbmAhy2jGDpzEiREkeucYJkipjZMkQ6tPGnQ7sxMjJI+lHSOoeCxCc3a9v2Ax4KWFa7vuUHfjB5x-uoAF3oiIEtjWkBth6oYRFGOBUkMzaDK66zaakKHwNwqjCehUCsIQ54GXZqS2BEDigic2y7EiADEAAs8BcgAHOgwVaeBqSZsJrTwDsSLRUCol9l0kFfj+ilwapaiIeq5kQTFCDxal0rJeJMrEvlmyqN2PHiR2b6tPUsx9PojCwAAajsxTFoMUmVml0Gwf+8Fqbl1aQHWYAAGTTWA7UAGLZEkhYANQlmWA2yelCmIkpKlAWNGmTbWM1ze1XXMcovWSZtFUTVNs3zbAS0yCt619XdRoPfWLlgPhKoA9MQP-UofGuXF5ikeREyUY6CX0SOy6rvRyjcBg2QxS6xlDP6XY9il+iDfJw3KaNOXnvlkDBjkoIjNUcJNdQLVdEUJSIFpuEQNpDWpPDJVsUjXTcbJqPo+gmMGWqwwSaZpUE+V30RZRNPUHTJywM1OxdCg9SNNCdzg-ab3Gj46h7Gx87IBEsRBns1pgy06CZvsowUrzUBFZxUCJZREY9t+XTsNwBDo-R4TFJckNIm04eaOJdvIILRIOAAwpEgdS5rzPa-obiMKYhZsywpD+RU9ZU2jGN6GAoY7LASeUWm7C9qRdemA3bEYJmACimaIpndxd9kfcD0JODVJROiOG39eNzO87ZKo0UT2xlnWZ5dmQM3re2W+9mOYKcJ-YGWbj8f++bNnLP6Cu8ALRGGB4DdpZ0N5Fs+HOACqb14JcibwEYKJdQhZpKQUrPISsO0IGVmOpAyA+0YGQHJrAGstxgBbQPAAWmgdgSsABSdgYB8EAE0wBIMgjg6CWDjpYAIUQ-ByhbikPIVpWsv1L6g33iDYGMYQY834tHAWftNAoXPtvZAoYhJsQjsoKOxV9C+xnLI225t57DhTunKI58NZa1alAfOhddYNDwKXcuysZxVwljXduncm7oBbh3WItj1GbG7qPMRsRJ4zkAh4we3iNEz2cXPS2S8sxeOtHyD2sUFE+3ov7Tx+hg6hyjDI+OwSO6uI9DoPA3sqxgELJQnaNDzhYP2jg7KsB6xsR3qYBauxYAcxwInGpDjewAHkvBpkIKoi2U9NEZx0UzG+Bi6iF2LogMxFdQnRHCRPSJ7tBGxN3PE0Rg8JFSLjsUDJdiZw5LyUrVpjjOnW26b0rJgYBnaMKnorohjCzGPwFMix9pF6zJXncB2loQCQiAA) double shifting does not seem to be a problem in vega.

Reply:
This doesn't lead to double shifting, but doesn't seem to help either.

```
{
  "data": {
    "values": [
      {"date": "Sun, 01 Jan 2012 23:00:00", "price": 150},
      {"date": "Sun, 02 Jan 2012 00:00:00", "price": 100},
      {"date": "Sun, 02 Jan 2012 01:00:00", "price": 170},
      {"date": "Sun, 02 Jan 2012 02:00:00", "price": 165},
      {"date": "Sun, 02 Jan 2012 03:00:00", "price": 200}
    ]
  },
  "mark": "line",
  "encoding": {
    "x": {
      "field": "date",
      "type": "temporal",
      "timeUnit": "utcyearmonthdatehoursminutes",
      "axis": {"labelAngle": 45},
      "scale": {"type": "time"} // changing to time or utc doesn't really cause any change?
    },
    "y": {"field": "price", "type": "quantitative"}
  }
}
```

Reply:
Ok actually, I'm wrong. It doesn't cause any change because we override the time format on the axis, and thus do not need to change the scale.  But it's still better to output the right thing, just in case users edit Vega and remove the format.